packages/wrangler/src/pipelines/index.ts

```diff
@@ -0,0 +1,459 @@
+				}
+				if (args["batch-max-rows"]) {
+					pipelineConfig.destination.batch.max_rows = args["batch-max-rows"];
+				}
+
+				let bucket = args.r2;
+				let accessKeyId = args["access-key-id"];
+				let secretAccessKey = args["secret-access-key"];
+				if (bucket || accessKeyId || secretAccessKey) {
+					const destination = pipelineConfig.destination;
+					if (bucket) {
+						pipelineConfig.destination.path.bucket = bucket;
+					}
+					destination.credentials = {
+						endpoint: getAccountR2Endpoint(accountId),
+						access_key_id: accessKeyId || "",
+						secret_access_key: secretAccessKey || "",
+					};
+					if (!accessKeyId && !secretAccessKey) {
+						await authorizeR2Bucket(
+							name,
+							accountId,
+							pipelineConfig.destination
+						);
+					}
+					if (!destination.credentials.access_key_id) {
+						throw new FatalError("Requires a r2 access key id");
+					}
+
+					if (!destination.credentials.secret_access_key) {
+						throw new FatalError("Requires a r2 secret access key");
+					}
+				}
+
+				if (args.authentication !== undefined) {
+					// strip off existing http source
+					pipelineConfig.source = pipelineConfig.source.filter(
+						(s) => s.type == "http"
+					);
+
+					// add back only if unauthenticated
+					if (!args.authentication) {
+						pipelineConfig.source.push({
+							type: "http",
+							format: "json",
+						});
+					}
+				}
+
+				if (args.transform !== undefined) {
+					pipelineConfig.transforms = [];
+					const split = args.transform.split(".");
+					if (split.length === 2) {
+						pipelineConfig.transforms.push({
+							script: split[0],
+							entrypoint: split[1],
+						});
+					} else if (split.length === 1) {
+						pipelineConfig.transforms.push({
+							script: split[0],
+							entrypoint: "Transform",
+						});
+					} else {
+						throw new Error(
+							"invalid transform: required syntax script.entrypoint"
+						);
+					}
+				}
+
+				if (args.filepath) {
+					pipelineConfig.destination.path.filepath = args.filepath;
+				}
+				if (args.filename) {
+					pipelineConfig.destination.path.filename = args.filename;
+				}
+
+				logger.log(`🌀 Updating pipeline "${name}"`);
+				const pipeline = await updatePipeline(accountId, name, pipelineConfig);
+				await metrics.sendMetricsEvent("update pipeline", {
+					sendMetrics: config.send_metrics,
+				});
+
+				logger.log(
+					`✅ Successfully updated pipeline "${pipeline.name}" with ID ${pipeline.id}\n`
+				);
+			}
+		)
+		.command(
+			"delete <pipeline>",
+			"Delete a pipeline",
+			(yargs) => {
+				return yargs.positional("pipeline", {
+					type: "string",
+					describe: "The name of the pipeline to delete",
+					demandOption: true,
+				});
+			},
+			async (args) => {
+				await printWranglerBanner();
+				const config = readConfig(args.config, args);
+				const accountId = await requireAuth(config);
+				const name = args.pipeline;
+
+				if (!name.match(/^[a-zA-Z0-9-]+$/)) {
```

Comment:
As this logic is repeated, it could be extracted to a shared function. This is good practice in general for Regex anyway as it's so notoriously unreadable (although this example is simple).